### PR TITLE
Use `with_bindgen` feature with `p256k1`

### PR DESCRIPTION
**Pull request by jcnelson**
This PR depends on [this PR against p256k1](https://github.com/Trust-Machines/p256k1/pull/74).  With this PR, `wsts` will continue to use `bindgen` to generate libsecp256k1 bindings for `p256k1` by default.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -21,7 +25,7 @@ num-traits = "0.2"
 polynomial = { version = "0.2.5", features = ["serde"] }
 primitive-types = "0.12"
 rand_core = "0.6"
-p256k1 = "5.4"
+p256k1 = { version = "5.4", default-features = false, git = "https://github.com/jcnelson/p256k1" }
```
Comment (xoloki) [inline Cargo.toml]: I just published `p256k1 5.5.0` with your bindgen changes, change the version here and get rid of the git pointer and we're good for this PR.

**Review by xoloki — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (jcnelson) [reply Cargo.toml]: Done :+1:

**Comment by jferrant** [comment]
Replaced by https://github.com/Trust-Machines/wsts/pull/31

**Comment by xoloki** [comment]
The merge from upstream ended up causing this PR to have a large diff, so I took the `Cargo.toml` changes and made a new PR.

**Pull request closed without merging**